---
string name
string config
string robot_origin